bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerRecoveryOp.java

```diff
@@ -40,22 +38,31 @@
  * the ledger at that entry.
  *
  */
-class LedgerRecoveryOp implements ReadCallback, AddCallback {
+class LedgerRecoveryOp implements ReadEntryListener, AddCallback {
+
     static final Logger LOG = LoggerFactory.getLogger(LedgerRecoveryOp.class);
-    LedgerHandle lh;
-    AtomicLong readCount, writeCount;
-    AtomicBoolean readDone;
-    AtomicBoolean callbackDone;
-    long entryToRead;
+
+    final LedgerHandle lh;
+    final AtomicLong readCount, writeCount;
+    final AtomicBoolean readDone;
```

Comment:
Minor thing, if we don't need `compareAndSet()` we can just use regular `volatile boolean` variables than don't require to be boxed